src/App.tsx

```diff
@@ -17,18 +52,34 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                setQuery={setQuery}
+                setFilter={setFilter}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length === 0 && <Loader />}
```

Comment:
will be better to handle it with additional boolean variable, that will define if data was loaded
for now if you'll receive an empty array from the server, user will see infinite loader on the page